src/hero/WebTab.js

```diff
@@ -0,0 +1,100 @@
+import {Component} from 'react';
+import React from 'react';
+import equal from 'deep-equal';
+import pretty from 'pretty';
+import {Prism as SyntaxHighlighter} from 'react-syntax-highlighter';
+import {prism} from 'react-syntax-highlighter/dist/esm/styles/prism';
+import {CopyToClipboard} from 'react-copy-to-clipboard';
+import {MDCRipple} from '@material/ripple/index';
+import {gtagCopyCode} from '../constants';
+
+import ReactGA from 'react-ga';
+
+const classesToRemove = [
+  ' mdc-ripple-upgraded--unbounded',
+  ' mdc-notched-outline--upgraded',
+  'hero-button ',
+  ' demo-card--hero',
+  'hero-top-app-bar',
+  ' style="transition-duration: 0s;"',
+  ' class=""'
+];
+
+const customStyle = {background: 'transparent'};
+
+export default class WebTab extends Component {
+  htmlRef = React.createRef();
+
+  state = {codeString: ''};
+  el = null;
+  ripples = [];
+
+  initRipple = (el) => {
+    if (el) {
+      const ripple = MDCRipple.attachTo(el);
+      ripple.unbounded = true;
+      this.ripples.push(ripple);
+    }
+  };
+
+  componentDidMount() {
+    this.initCodeString();
+  }
+
+  componentWillUnmount() {
+    this.ripples.map((ripple) => ripple.destroy());
+  }
+
+  componentDidUpdate(prevProps) {
```

Comment:
Maybe related to the not-pruning-when-already-in-web-tab bug: I'm noticing this gets called _3 times_ if an option is changed while the web tab is open. It only actually calls `initCodeString` on the 2nd call to this method.